src/controller.js

```diff
@@ -0,0 +1,111 @@
+const fs = require('fs');
+const path = require('path');
+
+const { task1: sort, task2: biggestPrice, task3 } = require('./tasks');
+
+let store = require('./inputData');
+
+function ok(res, body = { message: 'Ok' }) {
+  res.setHeader('Content-Type', 'application/json');
+  res.statusCode = 200;
+  res.write(JSON.stringify(body));
+  res.end();
+}
+
+function badRequest(res, message = { message: 'Bad request!' }) {
+  res.setHeader('Content-Type', 'application/json');
+  res.statusCode = 400;
+  res.write(JSON.stringify(message));
+  res.end();
+}
+
+function methodNotAllowed(res) {
+  res.setHeader('Content-Type', 'application/json');
+  res.statusCode = 405;
+  res.write(JSON.stringify({ message: 'Method not allowed!' }));
+  res.end();
+}
+
+function isIncorrectData(data) {
+  if (!Array.isArray(data) || data.length === 0) return true;
+
+  const incorrectArray = data.filter(
+    value =>
+      typeof value !== 'object' ||
+      !(
+        'type' in value &&
+        'color' in value &&
+        ('price' in value || 'priceForPair' in data[0])
+      ),
+  );
+  if (incorrectArray.length !== 0) return true;
+
+  return false;
+}
+
+module.exports.task1 = (request, response) => {
+  const { method, queryParams } = request;
+  const arrayClothes = store || [];
+
+  if (method !== 'GET') methodNotAllowed(response);
+  if (!queryParams.name && !queryParams.value) {
+    badRequest(response, { Message: 'No param!' });
+  }
+
+  let { value } = queryParams;
+  if (typeof Number(value) === 'number') value = Number(value);
```

Comment:
This condition will never return `false`
`typeof Number('put anything here')` is always `'number'`
You can use `Number.parseInt(variable, 10)` to check whether it's a number or not

Reply:
I try this `Number.parseInt(variable, 10)`, it is not work good. But it is: `!Number.isNaN(Number(value))`

Reply:
The way you updated it works too.
This is what I meant:
```
const str = '10';
const num = Number.parseInt(str, 10);
const isNumber = !Number.isNaN(num); // true (num = 10)
```

```
const str = 'red';
const num = Number.parseInt(str, 10);
const isNumber = !Number.isNaN(num); // false (num = NaN)
```

```
const str = '4you';
const num = Number.parseInt(str, 10);
const isNumber = !Number.isNaN(num); // true (num = 4)
```

Reply:
I don`t do so because 
```
const str = '4you';
const num = Number.parseInt(str, 10);
const isNumber = !Number.isNaN(num); // true (num = 4)
```

Why is this need do?